File: BarCode SDK/VB.NET/QR Code With Email/QrCodeEmailTemplate.vb
'*******************************************************************************************'
'                                                                                           '
' Get API Key https://app.pdf.co/signup                                                     '
' API Documentation: https://developer.pdf.co/api/index.html                                '
'                                                                                           '
' Note: Replace placeholder values in the code with your API Key                            '
' and file paths (if applicable)                                                            '
'                                                                      '
'*******************************************************************************************'


Public Class QrCodeEmailTemplate

#Region "Constructors"

    Public Sub New()
    End Sub

    Public Sub New(ByVal Email As String, ByVal Subject As String, ByVal Message As String)
        Me.Email = Email
        Me.Subject = Subject
        Me.Message = Message
    End Sub

#End Region


#Region "Properties"

    Public Property Email As String
    Public Property Subject As String
    Public Property Message As String

#End Region

#Region "Overloaded Methods"

    Public Overrides Function ToString() As String

        If String.IsNullOrEmpty(Email) AndAlso String.IsNullOrEmpty(Subject) AndAlso String.IsNullOrEmpty(Message) Then Return MyBase.ToString()

        Return $"MATMSG:TO:

{Email};

SUB:

{Subject}

BODY:

{Message}
;;"

    End Function

#End Region


End Class

File: BarCode SDK/C#/QR Code with Email/QrCodeEmailTemplate.cs
//*******************************************************************************************//
//                                                                                           //
// Get Your API Key: https://app.pdf.co/signup                                               //
// API Documentation: https://developer.pdf.co/api/index.html                                //
//                                                                                           //
// Note: Replace placeholder values in the code with your API Key                            //
// and file paths (if applicable)                                                            //
//                                                                                           //
//*******************************************************************************************//


using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CreateBarCode
{
    class QrCodeEmailTemplate
    {
        #region Constructors

        public QrCodeEmailTemplate() { }
        public QrCodeEmailTemplate(string Email, string Subject, string Message)
        {
            this.Email = Email;
            this.Subject = Subject;
            this.Message = Message;
        }

        #endregion

        #region Properties

        public string Email { get; set; }
        public string Subject { get; set; }
        public string Message { get; set; }

        #endregion

        #region Overloaded Methods

        public override string ToString()
        {
            if (string.IsNullOrEmpty(Email) && string.IsNullOrEmpty(Subject) && string.IsNullOrEmpty(Message))
                return base.ToString();

            return $@"MATMSG:TO:

{Email};

SUB:

{Subject}

BODY:

{Message}
;;";
        }

        #endregion
    }
}
